iris-mpc-cpu/src/next_gen_protocol/ng_worker.rs

```diff
@@ -282,6 +345,56 @@ pub async fn ng_replicated_is_match(
     Ok(opened.convert())
 }
 
+pub async fn gr_to_rep3(
+    session: &Session,
+    items: Vec<RingElement<u16>>,
+) -> eyre::Result<Vec<Share<u16>>> {
+    let network = session.network().clone();
+    let sid = session.session_id();
+    let next_party = session.next_identity()?;
+
+    // sending to the next party
+    let shares_a = items.clone();
+    network
+        .send(
+            NetworkValue::VecRing16(items).to_network(),
+            &next_party,
+            &sid,
+        )
+        .await?;
```

Comment:
This is missing the masking randomness, no? Either here, or already at the trick_dot, we need to add the REP3 correlated randomness to our own additive share, otherwise we leak a relation of our inputs to the dot product to the other party.

Reply:
I agree, and nicely spotted! I took inspiration from the `shamir_to_rep3` function and thought I didn't need to do something else.  Will add it in this PR

Reply:
fixed it in 4a98d904d2f7643707743ef0bbe52250b37619a8